connector/datadogconnector/connector.go

```diff
@@ -112,6 +130,27 @@ func (c *traceToMetricConnector) Capabilities() consumer.Capabilities {
 }
 
 func (c *traceToMetricConnector) ConsumeTraces(ctx context.Context, traces ptrace.Traces) error {
+	if c.enableContainerStats && len(c.containerTagAttrs) > 0 {
+		for i := 0; i < traces.ResourceSpans().Len(); i++ {
+			rs := traces.ResourceSpans().At(i)
+			attrs := rs.Resource().Attributes()
+			containerID, ok := attrs.Get(semconv.AttributeContainerID)
+			if !ok {
+				continue
+			}
+			ddContainerTags := attributes.ContainerTagsFromResourceAttributes(attrs)
+			for attr := range c.containerTagAttrs {
+				if val, ok := ddContainerTags[attr]; ok {
+					var cacheVal []string
+					if val, ok := c.containerTagCache.Get(containerID.AsString()); ok {
+						cacheVal = val.([]string)
```

Comment:
We probably want to do a map for the cache value so that we don't append the same container tag repeatedly?